src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -1561,24 +1565,51 @@ private void ClearTooltipText()
             ToolTipText = "";
         }
 
-        private void ClearPersistentWarning()
-        {
-            persistentWarning = String.Empty;
-        }
-
         /// <summary>
         /// Clears the errors/warnings that are generated when running the graph,
         /// the State will be set to ElementState.Dead.
         /// </summary>
         public virtual void ClearErrorsAndWarnings()
         {
             State = ElementState.Dead;
-            ClearPersistentWarning();
+            ClearPersistentWarnings();
 
             SetNodeStateBasedOnConnectionAndDefaults();
             ClearTooltipText();
         }
 
+        /// <summary>
+        /// Clears a specific persistent warning that was added before.
+        /// The State will remain as ElementState.PersistentWarning if other warnings remain.
+        /// </summary>
+        internal void ClearPersistentWarnings(string p = null)
+        {
+            if (State == ElementState.PersistentWarning)
+            {
+                if (p != null)
+                {
+                    persistentWarnings.Remove(p);
+                }
+                else
+                {
+                    persistentWarnings.Clear();
+                }
+
+                State = ElementState.Dead;
+
+                SetNodeStateBasedOnConnectionAndDefaults();
+
+                if (State == ElementState.PersistentWarning)
+                {
+                    ToolTipText = persistentWarning;
+                }
+                else
+                {
+                    ClearTooltipText();
+                }
```

Comment:
Why have you added all of this new logic?

Reply:
Because the NodeManipulator called ClearErrorsAndWarnings() before these changes.
ClearErrorsAndWarnings had a similar logic.

What I wanted to do is:
1. More than 1 persistent warning exists (i.e after clear is called other warnings remain)
    Keep the same state as before...only update the persistentWarning string and the Tooltip string
2. Single persistent warning (i.e after clear is called no other warnings remain)
   Try to find a valid state in which to put the Node's warning system.

Reply:
But all existing `ClearPersistentWarning` calls have _also_ been replaced with this new method and the old method didn't have this logic.

Reply:
Yep, a lot of redundant work.
I'll try a refactor